docs/source/developers/cpp/windows.rst

```diff
@@ -110,28 +170,42 @@ an out of source build by generating a MSVC solution:
    cd cpp
    mkdir build
    cd build
-   cmake .. -G "Visual Studio 14 2015 Win64" ^
+   cmake .. -G "Visual Studio 14 2015" -A x64 ^
          -DARROW_BUILD_TESTS=ON
    cmake --build . --config Release
 
+For newer versions of Visual Studio, specify the generator
+``Visual Studio 15 2017`` or ``Visual Studio 16 2019``.
+
 Building with Ninja and clcache
 ===============================
 
-The `Ninja <https://ninja-build.org/>`_ build system offsets better build
+The `Ninja <https://ninja-build.org/>`_ build system offers better build
 parallelization, and the optional `clcache
-<https://github.com/frerich/clcache/>`_ compiler cache which keeps track of
+<https://github.com/frerich/clcache/>`_ compiler cache keeps track of
 past compilations to avoid running them over and over again (in a way similar
 to the Unix-specific ``ccache``).
 
-Activate your conda build environment to first install those utilities:
+Newer versions of Visual Studio include Ninja. To see if your Visual Studio
+includes Ninja, run the initialization command shown above (``vcvarsall.bat``
```

Comment:
```suggestion
includes Ninja, run the initialization command shown :ref:`above <dev-windows-initialize>`(``vcvarsall.bat``
```

and then add that label before the "System Setup" section above as `.. _dev-windows-initialize:`

(I think  linking back here would help)

Reply:
Thanks! Added in 9ad6bdeb1ef1e7ea1159ab3bb038daeeb78872c5